cpp/dolfin/io/HDF5Interface.cpp

```diff
@@ -421,7 +430,7 @@ void HDF5Interface::set_mpi_atomicity(const hid_t hdf5_file_handle,
   herr_t status = H5Fset_mpi_atomicity(hdf5_file_handle, atomic);
   if (status == HDF5_FAIL)
```

Comment:
`if (status < 0)`

Reply:
I am confused with this request. `HDF5_FAIL` is defined in header, there are many uses throughout this file, why change just this one?

Reply:
The HDF5 docs simply say that the return code is negative in the case of any error, e.g. https://support.hdfgroup.org/HDF5/doc/RM/RM_H5F.html#File-SetMpiAtomicity, the so the check should respect this.